babel/src/main/java/org/apache/calcite/sql/babel/SqlColumnAttributeGeneratedMaxValue.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlWriter;
+
+/**
+ * A {@code SqlColumnAttributeGeneratedMaxValue} represents the MAXVALUE
+ * option of a GENERATED column attribute.
+ */
+public class SqlColumnAttributeGeneratedMaxValue extends
+    SqlColumnAttributeGeneratedOption {
+
+  public final SqlLiteral max;
+  public final boolean none;
+
+  /**
+   * Creates a {@code SqlColumnAttributeGeneratedMaxValue}.
+   *
+   * @param max     The amount specified in the MAXVALUE attribute.
+   * @param none    Whether NO MAXVALUE was specified.
+   */
+  public SqlColumnAttributeGeneratedMaxValue(SqlLiteral max, boolean none) {
+    this.max = max;
+    this.none = none;
+    assert none || max != null;
```

Comment:
I know I recommended using "assert" in a previous review. However, after speaking with some peers, the recommendation is to just use Objects.requireNonNull() for nullness assertion and use comments to indicate other assertions, rather than use "assert" statements.

Reply:
The issue is that here, "max" can be null, but only if "none" is true. Should I use Objects.requireNonNull() for "max", but enclose that in an if block so it only gets checked if "none" is false?

Reply:
Ah I see. Then in this particular situation maybe we should just describe in a comment and avoid the assert and requireNonNull() altogether.

Reply:
Ok. In that case I should probably add a null check in the unparse method (to avoid possibly getting a null pointer error when calling max.unparse() ), right?

Reply:
Yep, SGTM